### ISSUE TITLE
[x86] Failing CoreFX tests using x86

### ISSUE THREAD

**Issue by LLITCHEV** [dotnet/coreclr#4549]
These failures seems to be due to issues in the x86 Windows unwinder of CoreCLR.

D:\repos\corefx\Tools\tests.targets(154,5): error : One or more tests failed while running tests from 'System.Data.SqlClient.Tests' please check log for detail
s! [D:\repos\corefx\src\System.Data.SqlClient\tests\FunctionalTests\System.Data.SqlClient.Tests.csproj]
D:\repos\corefx\Tools\tests.targets(154,5): error : One or more tests failed while running tests from 'System.Runtime.Tests' please check log for details! [D:\
repos\corefx\src\System.Runtime\tests\System.Runtime.Tests.csproj]
D:\repos\corefx\Tools\tests.targets(154,5): error : One or more tests failed while running tests from 'System.IO.Compression.Tests' please check log for detail
s! [D:\repos\corefx\src\System.IO.Compression\tests\System.IO.Compression.Tests.csproj]
D:\repos\corefx\Tools\tests.targets(154,5): error : One or more tests failed while running tests from 'System.Reflection.TypeExtensions.Tests' please check log
 for details! [D:\repos\corefx\src\System.Reflection.TypeExtensions\tests\System.Reflection.TypeExtensions.Tests.csproj]
D:\repos\corefx\Tools\tests.targets(154,5): error : One or more tests failed while running tests from 'System.Runtime.Numerics.Tests' please check log for deta
ils! [D:\repos\corefx\src\System.Runtime.Numerics\tests\System.Runtime.Numerics.Tests.csproj]
D:\repos\corefx\Tools\tests.targets(154,5): error : One or more tests failed while running tests from 'System.Runtime.Extensions.Tests' please check log for de
tails! [D:\repos\corefx\src\System.Runtime.Extensions\tests\System.Runtime.Extensions.Tests.csproj]
D:\repos\corefx\Tools\tests.targets(154,5): error : One or more tests failed while running tests from 'System.IO.Packaging.Tests' please check log for details!
 [D:\repos\corefx\src\System.IO.Packaging\tests\System.IO.Packaging.Tests.csproj]
D:\repos\corefx\Tools\tests.targets(154,5): error : One or more tests failed while running tests from 'System.IO.Pipes.Tests' please check log for details! [D:
\repos\corefx\src\System.IO.Pipes\tests\System.IO.Pipes.Tests.csproj]
D:\repos\corefx\Tools\tests.targets(154,5): error : One or more tests failed while running tests from 'System.IO.Compression.ZipFile.Tests' please check log fo
r details! [D:\repos\corefx\src\System.IO.Compression.ZipFile\tests\System.IO.Compression.ZipFile.Tests.csproj]
D:\repos\corefx\Tools\tests.targets(154,5): error : One or more tests failed while running tests from 'System.Reflection.Tests' please check log for details! [
D:\repos\corefx\src\System.Reflection\tests\System.Reflection.Tests.csproj]

**Comment by LLITCHEV** [comment]
It looks like this is a x86 unwinding problem. It looks like it doesn't know how to properly unwind "bnd jmp/call" instructions from the Intel's MPX support.

**Comment by LLITCHEV** [comment]
Seems to be related to this commit:
https://github.com/dotnet/coreclr/commit/3c03a0f65c23565a1799c138470d952a039c142e

**Comment by RussKeldorph** [comment]
@janvorli @sergiy-k FYI--Not sure you saw this due to RTM milestone.  If you've already triaged it, my apologies.

**Comment by janvorli** [comment]
@LLITCHEV I am unable to repro the issue. Can you please tell me the exact steps to repro it?

**Comment by janvorli** [comment]
Minimal repro:
```C#
using System;  

internal class Outside
{
    public class Inside { }
}

internal class Outside<T>
{
    public class Inside<U> { }
}

namespace Test
{
     class Program
    {
        static void Main(string[] args)
        {
            try
            {
                Type.GetType("Outside`1[System.Boolean, System.Int32]", throwOnError: true, ignoreCase: false);
            }
            catch (ArgumentException ex)
            {
                Console.WriteLine("Caught correct exception");
            }
        }
    }
}
```

**Comment by janvorli** [comment]
Fixed by #4830
